sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -242,15 +242,17 @@ export class ServiceBusClient {
    * Creates a Sender which can be used to send messages, schedule messages to be sent at a later time
    * and cancel such scheduled messages.
    */
-  createSender(queueOrTopicName: string): Sender {
+  async createSender(queueOrTopicName: string): Promise<Sender> {
```

Comment:
Should the docs be updated to mention that the sender will be automatically opened when created? This does mean they'd need to explicitly close the sender (or client) if they didn't want it to keep the process open anymore.

Reply:
This is interesting. I mean, technically there was no reason for them not to call `close()` before but, as you're pointing out, they didn't have to. 

I can update the doc comment (not sure what I'll write there yet?)

Reply:
Maybe it's not super important if we treat this as a new API and that's the behavior (we can mention it in a migration guide for existing users). But it might be worth saying something about the fact that a connection with the service is established if the motivation for this change is that customers want to know if the connection is open or not.

Reply:
Yes, adding it to the migration guide is a good idea.

Reply:
Migration guide updated and, thanks to you, I remembered to update the samples as well!